src/entrystore/src/seg/memcache.rs

```diff
@@ -285,19 +285,34 @@ impl Storage for Seg {
     }
 
     fn cas(&mut self, cas: &Cas) -> Response {
-        // duration of zero is treated as no expiry. as we have
-        // no way of checking the cas value without performing a cas
-        // and checking the result, setting the shortest possible ttl
-        // results in nearly immediate expiry
-        let ttl = cas.ttl().get().unwrap_or(1);
+        // TTL of None means that it doesn't expire. In `Seg` storage
+        // a TTL of zero maps to the longest TTL representable which
+        // is ~97 days.
+
+        let ttl = cas.ttl().get().unwrap_or(0);
+
+        // Since we allow specifying Unix timestamps as TTLs, we can actually
+        // make a request that says that if the CAS value matches, the item
+        // should be immediately expired.
+        //
+        // However, we cannot check the CAS value without attempting to store,
```

Comment:
The behavior about Unix timestamp isn't unique to cas, right? So if we have a `set` or `replace` command, it could also be that we should immediately delete the key from underlying storage. Today we handle them individually in the `memcache` entrystore implementation. However, now I wonder if the better place to deal with that is actually in `fn insert` in the seg module. Basically, if we handle `ttl == 0` as delete, and collapse `ttl <= 0` to `ttl == 0`, we can keep the data type signature but also simplify the entrystore implementation.

Reply:
That's a legit critique. Can look at making that change

Reply:
we should probably merge this PR first if we think it's not risky because we're left with the choice of either failing builds non-deterministically or not running integration tests in CI